openpdf/src/main/java/com/lowagie/text/pdf/PdfPKCS7.java

```diff
@@ -556,17 +571,212 @@ public PdfPKCS7(byte[] contentsKey, String provider) {
+   * 
+   * https://tools.ietf.org/rfc/rfc8017.txt ->  A.2.3.  RSASSA-PSS Page 60
+   * 
+   * Example of a structure with all values explicitly defined:
+   * DER Tagged Object  : | D
+   * Sequence           : | | +
+   * Sequence           : | | | +
+   * Sequence           : | | | | +
+   * DER Tagged Object  : | | | | | D
+   * integer            : | | | | | | 2
+   * integer            : | | | | | 123456789
+   * Sequence           : | | | | | +
+   * ObjectIdentifier   : | | | | | | 1.2.840.113549.1.1.10  -- RSASSA-PSS
+   * Sequence           : | | | | | | +
+   * DER Tagged Object  : | | | | | | | D
+   * Sequence           : | | | | | | | | +
+   * ObjectIdentifier   : | | | | | | | | | 2.16.840.1.101.3.4.2.1  -- SHA256
+   * null               : | | | | | | | | | N
+   * DER Tagged Object  : | | | | | | | D
+   * Sequence           : | | | | | | | | +
+   * ObjectIdentifier   : | | | | | | | | | 1.2.840.113549.1.1.8  -- MGF1
+   * Sequence           : | | | | | | | | | +
+   * ObjectIdentifier   : | | | | | | | | | | 2.16.840.1.101.3.4.2.1  -- SHA256
+   * null               : | | | | | | | | | | N
+   * DER Tagged Object  : | | | | | | | D
+   * integer            : | | | | | | | | 32  -- Salt length 32
+   *
+   * @param possibleSignatureInformation probably the signerinformation object.
+   * @return the PSSParameterSpec if this is a signature using RSAASS/PSS, null if not. If not all values are
+   * specifically defined, the default values are used as described in RFC8017
+   */
+    private PSSParameterSpec getRSASSAPSSParameterSpec(Object possibleSignatureInformation)
+    {
+      if (!(possibleSignatureInformation instanceof DERTaggedObject))
+      {
+        return null;
+      }
+      DERTaggedObject d = (DERTaggedObject)possibleSignatureInformation;
+
+      if (!(d.getObject() instanceof ASN1Sequence))
+      {
+        return null;
+      }
+      ASN1Sequence ds = (ASN1Sequence)d.getObject();
+
+      if (!(ds.getObjectAt(0) instanceof ASN1Sequence))
+      {
+        return null;
+      }
+      ASN1Sequence dss = (ASN1Sequence)ds.getObjectAt(0);
+
+      if (!(dss.getObjectAt(0) instanceof ASN1Sequence))
+      {
+        return null;
+      }
+      ASN1Sequence dsss = (ASN1Sequence)dss.getObjectAt(0);
+
+      if (!(dsss.getObjectAt(2) instanceof ASN1Sequence))
+      {
+        return null;
+      }
+      ASN1Sequence dssss = (ASN1Sequence)dsss.getObjectAt(2);
+
+      if (!(dssss.getObjectAt(0) instanceof ASN1ObjectIdentifier))
+      {
+        return null;
+      }
+      ASN1ObjectIdentifier dsssso = (ASN1ObjectIdentifier)dssss.getObjectAt(0);
+
+      String encAlgOid = dsssso.getId();
+
+      if (!"1.2.840.113549.1.1.10".equals(encAlgOid))
+      { // not RSASSA-PSS, so no PSSParameterSpec to be returned
+        return null;
+      }
+
+      // defaults as defined in https://tools.ietf.org/rfc/rfc8017.txt
+      String digestAlgOid = "1.3.14.3.2.26";                      // SHA1
```

Comment:
Diese Zuweisung ist unnötig, da der Wert entweder 1) nicht verwendet wird, weil ein vorzeitiges Return stattfindet, oder 2) in Zeile 704 überschrieben wird.